### PR TITLE
ECPf wrap-up

### LINKED ISSUE #7452: driver-only ECC: ECPf wrap-up
- [x] add ChangeLog entries
- [x] update any existing documentation that needs updating (strategy doc?) - edit: didn't need updating now, will revisit before 3.5 as part of #7757
- [x] create specific document about driver-only builds (see below)

We should have some user documentation about driver-only builds in a place that's easy to find, perhaps `docs/driver-only-builds.md`.

It should clearly document what's supported and what's not, and what the requirements and limitations are. Non-exhaustive list:
- tell that hashes are OK, ECC mostly and ciphers not at all
- document the need to call `psa_crypto_init()` before using accelerated operations in driver-only builds
- document the current state of ECC support: Bignum still needed, and part of ecp.c with some options.

### PR DESCRIPTION
**Pull request by mpg**
## Description

Resolves #7452. The most important part is adding some user-oriented documentation about driver-only builds. So far, only covering generalities as well as ECC - the section on hashes is overdue but will be added in another PR.

## PR checklist

- [x] **changelog** provided
- [x] **backport** not required - documenting 3.x-only features
- [x] **tests** not required - documentation only

### CODE REVIEW

File: docs/driver-only-builds.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+Supported means that when those are provided only by drivers, everything
+(including PK, X.509 and TLS if `MBEDTLS_USE_PSA_CRYPTO` is enabled) should
+work in the same way as if the mechanisms where built-in, except as documented
+in the "Limitations" sub-sections of the sections dedicated to each family
+below.
+
+In the near future (end of 2023), we are planning to also add support for
+ciphers (AES) and AEADs (GCM, CCM, ChachaPoly).
+
+Currently (mid-2023) we don't have plans to extend this to RSA of FFDH. If
```
Comment (valeriosetti) [inline docs/driver-only-builds.md:67]: Maybe I'm wrong, but I think that FFDH is actually supported thanks to the work done by Przemyslaw. Isn't it?

File: docs/driver-only-builds.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  public part is in compressed format;
+- `MBEDTLS_PK_PARSE_EC_EXTENDED` - support for parsing ECC keys where the
+  curve is identified not by name, but by explicit parameters;
+- `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` - support for deterministic
+  derivation of an ECC keypair with `psa_key_derivation_output_key()`.
+
+Note: when one of the above options is enabled, you can still `MBEDTLS_ECP_C`
+in `mbedtls_config.h`, and it will still result in some code size savings, but
+not as much as when none of these are enabled, as a subset of `ecp.c` will
+still be included in the build in order to support these.
```
Comment (valeriosetti) [inline docs/driver-only-builds.md:123]: Maybe it's me, but I would rephrase it as follows: 

```suggestion
Note: when any of the above options is enabled a subset of `ecp.c` will
automatically be included in the build in order to support it. Therefore
you can still disable `MBEDTLS_ECP_C` in `mbedtls_config.h` and this will
result in some code size savings, but not as much as when none of the
above features are enabled.
```

File: docs/driver-only-builds.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+We do have plans to support each of these with `ecp.c` fully removed in the
+future, however no established timeline. If you're interested, please let us
+know, so we can take it into consideration in our planning.
+
+### Limitations regarding restartable / interruptible ECC operations
+
+At the moment, the is not driver support for interruptible operations
```
Comment (valeriosetti) [inline docs/driver-only-builds.md:131]:
```suggestion
At the moment, there is not driver support for interruptible operations
```

**Review by valeriosetti — COMMENTED** [review]
Only one major question about FFDH. The rest are just a typo and a potential rephrase ;)

File: docs/driver-only-builds.md (hunk 1 above)
Comment (mpg) [reply docs/driver-only-builds.md:67]: You're absolutely right, there's a PR with "FFDH driver-only" in its title, which I reviewed and merged, not sure how I could have missed that! Probably because we did it more for uniformity with the ECDH case, to ease maintenance by avoiding special cases / half-done things, than because there was real demand for it (as opposed to ECC), so my brain didn't connect the dots. Thanks for the catch, I'll fix this!

**Comment by mpg** [comment]
@valeriosetti Thanks for your review, especially catching the wrong statement about FFDH. I addressed your feedback, please review again.

**Review by valeriosetti — APPROVED** [review]
It looks good to me now ;)

**Review by tom-daubney-arm — CHANGES_REQUESTED** [review]
Overall looks really good and is clear but there are a few places where I feel the clarity could be improved a little bit as per my comments.

**Comment by mpg** [comment]
@tom-daubney-arm @paul-elliott-arm Thanks for your reviews! I believe I've addressed your feedback and this is ready for the next round.

**Comment by mpg** [comment]
@valeriosetti Can you re-review this, since Paul won't get a chance before some time?

**Review by valeriosetti — APPROVED** [review]
Sorry I forgot to review latest changes. They look good to me!
